### PR TITLE
Adding downstream support to CRD

### LINKED ISSUE #2886: Add support for downstream entries in the Spiffe ID CRD
* **Version**: Current
* **Platform**: Kubernetes
* **Subsystem**: CRD Registrar

While using the CRD resource as a means of registering secondary (nested) servers in other Kubernetes clusters, I found that there is not currently support for adding downstream identities via the CRD. I think this is a valid use case for the CRD.

### PR DESCRIPTION
**Pull request by noxora**
**Pull Request check list**

- [x] Commit conforms to CONTRIBUTING.md?
- [x] Proper tests/regressions included?
- [x] Documentation updated?

**Affected functionality**

This PR impacts the Spiffe ID CRD resource and the controller in charge of reconciling the Spiffe ID resource

**Description of change**

This PR adds a field to the CRD which allows the user to mark resource as a downstream entry. 

**Which issue this PR fixes**

fixes #2886

### CODE REVIEW

File: doc/plugin_server_nodeattestor_x509pop.md [hunk 1]
```diff
@@ -41,3 +41,17 @@ A sample configuration:
...
+
+Some useful values are:
+
+| Value                 | Description                                                  |
+| ----------------------| ------------------------------------------------------------ |
+| .PluginName           | The name of the plugin                                       |
+| .Fingerprint          | The SHA1 fingerprint of the agent's x509 certificate         |
+| .TrustDomain          | The configured trust domain                                  |
+| .Subject.CommonName   | The common name field of the agent's x509 certificate        |
+| .Subject.SerialNumber | The serial number field of the agent's x509 certificate      |
```
Comment (faisal-memon) [inline doc/plugin_server_nodeattestor_x509pop.md:57]: Is this section relevant to this PR?

**Comment by faisal-memon** [comment]
@noxora Can you update the README and add a section on using the new field? https://github.com/spiffe/spire/tree/main/support/k8s/k8s-workload-registrar/mode-crd

File: doc/plugin_server_nodeattestor_x509pop.md (hunk 1 above)
Comment (noxora) [reply doc/plugin_server_nodeattestor_x509pop.md:57]: This isn't. This is from https://github.com/spiffe/spire/pull/2851 and some confusion from me. I haven't used forks in a bit and am confused because this documentation change should already be merged into main per the linked PR.

**Comment by noxora** [comment]
> @noxora Can you update the README and add a section on using the new field? Or just a simple update to the SPIFFE ID Custom Resource Example section? https://github.com/spiffe/spire/blob/main/support/k8s/k8s-workload-registrar/mode-crd/README.md

@faisal-memon I've gone ahead and added to the example, I don't think it's complicated enough to warrant its own section, since it's just adding a boolean field to the CRD, but it's a good idea to clarify that it's both optional and should default to false

Outside of that, I wanted to test my changes in an environment and make sure that there isn't a nil pointer exception because I'm worried about support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1/spiffeid_types.go being nil and then attempting to set nil when creating the spiffeID

File: doc/plugin_server_nodeattestor_x509pop.md (hunk 1 above)
Comment (faisal-memon) [reply doc/plugin_server_nodeattestor_x509pop.md:57]: Did you pull this branch while still on the other branch?

Comment (noxora) [reply doc/plugin_server_nodeattestor_x509pop.md]: I did, I fixed it

**Comment by faisal-memon** [comment]
> 
> Outside of that, I wanted to test my changes in an environment and make sure that there isn't a nil pointer exception because I'm worried about support/k8s/k8s-workload-registrar/mode-crd/api/spiffeid/v1beta1/spiffeid_types.go being nil and then attempting to set nil when creating the spiffeID

Are you worried the new field will be nil?

**Comment by faisal-memon** [comment]
The changes look good for manually created entries. Do we need a mechanism for auto-generated entries to have `downstream` set?

**Comment by noxora** [comment]
Yes, I was worried it would be nil. I don't think this change would impact auto-registration unless that needed to set downstream in the yaml. My other concerns were that there might be other behavior that needs to be updated that I hadn't addressed, since I don't know the entire system very well, just what I've read. 

By manually created entries, do you mean those created with yaml files or those entered through a terminal on the server?

**Review by azdagron — APPROVED** [review]
\o/, thanks @noxora
